### PR TITLE
[feature] Limit length of component titles

### PR DESCRIPTION
**Pull request by geeksnglitter**
Purpose
-------------------------
Currently when making a new component on the osf a modal will pop up and allow you to enter a title as long as you wish, but if your title is longer than 200 characters, the component will not be created.

Closes Issue https://github.com/CenterForOpenScience/openscienceframework.org/issues/2993

Changes
------------------
I simply added a maxlength="200" into the field for component titles. Stopping users form being able to enter a title longer than 200 characters.

Edit: Because we want there to be consistency in title length, there were many changes in order to secure that title length remains around 200 throughout the system. Both front end and back end validations were added in order to enforce the 200 character limit on titles. You can no longer edit a title to be as long as you desire as shown in https://github.com/CenterForOpenScience/osf.io/issues/2993#issuecomment-110046427.

Projects will now not be able to be created if a user attempts to make a title past 200 characters. They will also not be able to enter titles past 200 characters in the fields that edit said titles, so project creation should not fail for this reason.

### CODE REVIEW

**Comment by lyndsysimon** [comment]
Replaces #3005

**Comment by lyndsysimon** [comment]
Note that this PR has been reviewed by both Saman and me. Pending Travis, this is ready for you @sloria.

**Comment by sloria** [comment]
This seems to have reverted a hotfix. The commit is https://github.com/CenterForOpenScience/osf.io/commit/f17098f11b17cfde09f1a837e19250e791e08401 . @lyndsysimon @geeksnglitter Can you look into this?

**Comment by lyndsysimon** [comment]
It looks like the regression was part of a difficult merge - it's been fixed, and should be good to go!

File: website/project/model.py [hunk 1]
```diff
@@ -515,12 +514,15 @@ def validate_category(value):
...
-    """Validator for Node#title. Makes sure that the value exists.
+    """Validator for Node#title. Makes sure that the value exists and is not
+    above 200 characters.
     """
     if value is None or not value.strip():
         raise ValidationValueError('Title cannot be blank.')
-    return True
 
+    if len(value) > 200:
+            raise ValidationValueError('Title cannot exceed 200 characters.')
```
Comment (sloria) [inline website/project/model.py]: This appears to be over-indented.

Comment (sloria) [inline website/project/model.py]: Fixed when I merged.

**Pull request merged**